### DISCUSSION TITLE
Is there a way to set multiple database fields from one dropdown

### DISCUSSION THREAD

**Discussion by geckigek** [netbox-community/netbox#15038] [Help Wanted!] [answered]
Hi there,

I am currently working on a ptp plugin for netbox. The data structure looks as follows:
```
class PTPProfilesChoices(ChoiceSet):

    CHOICES = [
        (0, 'SMPTE-2059-2'),
        (1, 'AES67'),
        (2, 'default'),
    ]

class PTPSyncIntChoices(ChoiceSet):

    CHOICES = [
        (-3, '8 per second'),
        (-2, '4 per second'),
        (-1, '2 per second'),
        (0, '1 per second'),
        (1, '1/2 per second'),
    ]

class PTPAnnounceIntChoices(ChoiceSet):

    CHOICES = [
        (-3, '8 per second'),
        (-2, '4 per second'),
        (-1, '2 per second'),
        (0, '1 per second'),
        (1, '1/2 per second'),
    ]

class PTPDelayReqIntChoices(ChoiceSet):

    CHOICES = [
        (-3, '8 per second'),
        (-2, '4 per second'),
        (-1, '2 per second'),
        (0, '1 per second'),
        (1, '1/2 per second'),
    ]

class PTPTemplate(NetBoxModel):

    name = models.CharField(
        max_length = 30
    )

    devicetype = models.OneToOneField(
        to = 'dcim.DeviceType',
        on_delete=models.CASCADE,
        blank = False,
    )

    ptp_enabled = models.BooleanField(
        default= False
    )

    mode = models.CharField(
        max_length = 30,
        choices = PTPModeChoices
    )

    ptp_domain = models.IntegerField(
        default = 0
    )

    log_ptp_sync_interval = models.IntegerField(
        default = -3,
        choices = PTPSyncIntChoices,
    )

    log_ptp_announce_interval = models.IntegerField(
        default = 0,
        choices = PTPAnnounceIntChoices,
    )

    log_ptp_delayreq_interval = models.IntegerField(
        default = 0,
        choices = PTPDelayReqIntChoices,

    ptp_profile = models.IntegerField(
        default = 0,
        choices = PTPProfilesChoices,
    )
```

So my requirement would be to set the value of ptp_profile in a rendered drop down menu and when I set hte ptp_profile value the values of log_ptp_sync_interval, log_ptp_announce_interval and log_ptp_delayreq_interval will automaticly set to a pre defined value from the ptp profile. For Example if ptp_profile would be 0 (SMPTE-2059-2) the three values should be set accordingly:

log_ptp_sync_interval = -3
log_ptp_announce_interval = 0
log_ptp_delayreq_interval = -3

Is this even possible and is there a way to achieve such a behavior?

Best regards

Stefan

**Comment by candlerb** [comment]
Aside: please can you edit your post and put three backticks (` ``` `) on a line of their own, before and after the code, to make it easier to read.

> So my requirement would be to set the value of ptp_profile in a rendered drop down menu and when I set hte ptp_profile value the values of log_ptp_sync_interval, log_ptp_announce_interval and log_ptp_delayreq_interval will automaticly set to a pre defined value from the ptp profile.

If you want to do this in the model, you can just hook into the `save()` or `clean()` methods. There are plenty of examples of this in Netbox's own code. Maybe something like this outline:

```
class PTPTemplate(NetBoxModel):
    ...
    def save(self, *args, **kwargs):
        if self.log_ptp_sync_interval is None:
            self.log_ptp_sync_interval = ... choice based on self.ptp_profile ...
        if self.log_ptp_announce_interval is None:
            self.log_ptp_announce_interval = ... choice based on self.ptp_profile ...
        ... etc

        return super().save(*args, **kwargs)
```

(This approach would mean not setting an explicit default value of 0, and instead letting the None value propagate so that `save()` can detect when the user didn't make a selection. Or you could have a dummy default value like -32768 to indicate that no choice was made)

If you are talking about a form displayed on-screen, and you want a click on one selection menu to trigger updates to other selection menus, then that's harder - at that point you're messing about with Javascript.

**Reply by candlerb** [reply, marked as answer]
There is a concrete example of this in the Interface wifi attributes, in `netbox/dcim/models/device_components.py`

```
    rf_channel = models.CharField(
        max_length=50,
        choices=WirelessChannelChoices,
        blank=True,
        verbose_name=_('wireless channel')
    )
    rf_channel_frequency = models.DecimalField(
        max_digits=7,
        decimal_places=2,
        blank=True,
        null=True,
        verbose_name=_('channel frequency (MHz)'),
        help_text=_("Populated by selected channel (if set)")
    )
    rf_channel_width = models.DecimalField(
        max_digits=7,
        decimal_places=3,
        blank=True,
        null=True,
        verbose_name=('channel width (MHz)'),
        help_text=_("Populated by selected channel (if set)")
    )
...

    def save(self, *args, **kwargs):

        # Set absolute channel attributes from selected options
        if self.rf_channel and not self.rf_channel_frequency:
            self.rf_channel_frequency = get_channel_attr(self.rf_channel, 'frequency')
        if self.rf_channel and not self.rf_channel_width:
            self.rf_channel_width = get_channel_attr(self.rf_channel, 'width')

        super().save(*args, **kwargs)
```

**Reply by geckigek** [reply]
Thanks for your help @candlerb!

The example of yours worked out fine! Thank you!
